### PR TITLE
add github actions workflow and jest setup

### PR DESCRIPTION
**Pull request by LucileTech**
## I'm submitting a ...
- feature request

## What is the current behavior?
simple tests and setting up a workflow for them 

## What is the expected behavior?
simple tests and setting up a workflow for them 

## What is the motivation / use case for changing the behavior?
#52

## Checklist

- [x] tests are present (if necessary)
- [x] no commented out code
- [x] no debbugging code
- [x] errors are handled
- [x] properly typed

### CODE REVIEW

**Comment by LucileTech** [comment]
@huyfififi would you have some insights on how I could achieve this task?

**Comment by huyfififi** [comment]
@LucileTech We need to specify `cache-dependency-path` to allow a GitHub workflow to locate the right dependencies lock file as our directory structure seems not to be supported.
You can check how the option is used in `test-backend.yml`.

Also, there is the latest version of `setup-node@v4`. We might want to consider using it.

https://github.com/actions/setup-node

**Comment by LucileTech** [comment]
> @LucileTech We need to specify `cache-dependency-path` to allow a GitHub workflow to locate the right dependencies lock file as our directory structure seems not to be supported. You can check how the option is used in `test-backend.yml`.
> 
> Also, there is the latest version of `setup-node@v4`. We might want to consider using it.
> 
> https://github.com/actions/setup-node

Thank you for your help about the cache-dependency-path!!  The CI is now going through all checks. I added a very basic jest test, I was thinking about adding more in an other Pull Request.

File: frontend/src/__tests__/weeklycalendar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+test("renders all day buttons", () => {
+  // Render the WeeklyCalendar component
+  render(<WeeklyCalendar />);
+
+  // Check if all day buttons are rendered
+  const daysOfWeek = ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"];
+  daysOfWeek.forEach((day) => {
+    const dayButton = screen.getByRole("button", { name: new RegExp(day) });
+    expect(dayButton).toBeInTheDocument();
```
Comment (huyfififi) [inline frontend/src/__tests__/weeklycalendar.test.tsx:13]: 👍

**Review by huyfififi — APPROVED** [review]
👍

**Pull request merged**